site/app/libraries/Access.php

```diff
@@ -914,8 +914,16 @@ public function resolveDirPath(string $dir, string $path) {
         }
         $path = implode(DIRECTORY_SEPARATOR, $parts);
 
-        //Make sure it starts with the dir base
-        if (!str_starts_with($path, $info["base"])) {
+        // Diffferent base path for downloading output.csv of rainbow grades
+        if ($dir === "rainbow_grades" && substr($path, -11) === "/output.csv") {
+            // Removing "/upload" from path
+            $base = substr($info["base"], 0, -23) . "/rainbow_grades";
+            //checking the path
+            if (!str_starts_with($path, $base)) {
+                $path = $base . $path;
+            }
+        }//Make sure it starts with the dir base
+        elseif (!str_starts_with($path, $info["base"])) {
```

Comment:
Sorry, I'm actually going to go back on what I said. I thought the Access.php rainbow_grades directory pointed to this one but didn't realize it had the upload on it. I don't think we should add something like this to Access.php. If you could just grab your code from before in the git log that had the new route which downloaded the file, that would be great.

Reply:
Fixed in https://github.com/Submitty/Submitty/pull/10079/commits/7fcf166721524d006e8da11a8feb43b52365c2fb